pandas/io/excel/_odfreader.py

```diff
@@ -102,7 +102,11 @@ def get_sheet_data(self, sheet, convert_float: bool) -> list[list[Scalar]]:
         table: list[list[Scalar]] = []
 
         for sheet_row in sheet_rows:
-            sheet_cells = [x for x in sheet_row.childNodes if x.qname in cell_names]
+            sheet_cells = [
+                x
+                for x in sheet_row.childNodes
+                if "qname" in dir(x) and x.qname in cell_names
```

Comment:
Can we do getattr(x, "qname")?

Reply:
@phofl The tests are passing only with `if "qname" in dir(x) and getattr(x, "qname") in cell_names`. But I don't know if that's what you meant.

Reply:
use `hasattr` not `dir`